azure-core/azure/core/credentials.py

```diff
@@ -0,0 +1,16 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import Iterable  # pylint:disable=unused-import
+from typing_extensions import Protocol
+
+
+class SupportsGetToken(Protocol):
+    """Protocol for classes able to provide OAuth tokens"""
+
+    # pylint:disable=too-few-public-methods
```

Comment:
Who owns the linter configuration? I would liker to raise the idea of turning this check off.

Reply:
I'd guess @lmazuel ?  Close relative `too-many-public-methods` is another we could probably do without.